simpleclient/src/test/java/io/prometheus/client/SimpleCollectorTest.java

```diff
@@ -145,6 +145,15 @@ public void testReservedLabelNameThrows() {
     Gauge.build().name("a").labelNames("__name__").help("h").create();
   }
 
+  @Test
+  public void testUnitsAdded() {
+    Gauge g = Gauge.build().name("a").unit("seconds").help("h").create();
+    assertEquals("a_seconds", g.fullname);
+
+    Gauge g2 = Gauge.build().name("a_seconds").unit("seconds").help("h").create();
+    assertEquals("a_seconds", g2.fullname);
```

Comment:
Worth adding a test to observe the unit being added if the name doesn't correctly have the unit on it already when built?

Reply:
That's what the test right above this is doing.